pkg/config/config.go

```diff
@@ -47,13 +33,13 @@ func (dc *DaemonConfig) ValidateConfig() error {
 		return fmt.Errorf("ValidateConfig(): invalid \"PeriodicUpdate\" value %v", dc.PeriodicUpdate)
 	}
 
-	if dc.SubnetManager.Plugin == "" {
+	if dc.Plugin == "" {
 		return fmt.Errorf("ValidateConfig(): no plugin selected, supported plugins %v", supportedPlugins)
 	}
 
 	if !dc.isSupportedPlugin() {
```

Comment:
If its intended for customers to provide their own, we should not enforce against a fixed list